File: backend/src/main/kotlin/org/globe42/storage/ReadableFile.java
package org.globe42.storage;

import java.io.InputStream;
import java.nio.channels.Channels;

import com.google.cloud.storage.Blob;

/**
 * A file that can be read
 * @author JB Nizet
 */
public class ReadableFile {
    private final FileDTO file;
    private final Blob blob;

    public ReadableFile(Blob blob, String prefix) {
        this.blob = blob;
        this.file = new FileDTO(blob, prefix);
    }

    public InputStream getInputStream() {
        return Channels.newInputStream(this.blob.reader());
    }

    public FileDTO getFile() {
        return file;
    }
}

File: backend/src/main/kotlin/org/globe42/storage/ReadableFile.kt
package org.globe42.storage

import com.google.cloud.storage.Blob
import java.io.InputStream
import java.nio.channels.Channels

/**
 * A file that can be read
 * @author JB Nizet
 */
open class ReadableFile(private val blob: Blob, prefix: String) {
    open val file: FileDTO = FileDTO(blob, prefix)

    open val inputStream: InputStream
        get() = Channels.newInputStream(this.blob.reader())
}
